### ISSUE TITLE
Unnecessary warning introduced by recent PR

### ISSUE THREAD

**Issue by spickermann** [googleapis/google-cloud-ruby#22896]
This recent https://github.com/googleapis/google-cloud-ruby/pull/22866 enabled setting the `endpoint` configuration option for  `Google::Cloud.bigquery` by an `ENV["BIGQUERY_EMULATOR_HOST"]` environment variable. 

At it first glance, this change shouldn't have caused any issues, because when `ENV["BIGQUERY_EMULATOR_HOST"]` is not set, it still returns `nil` like the previous default. But the validator works slightly different when `add_field!` is called with a `Google::Cloud::Config.deferred` block as `initial` value instead of a simple `nil` – even if the block returns `nil` when called.

Previously, when the initial value for `endpoint` was explicitly set to `nil` in [this line](https://github.com/googleapis/google-cloud-ruby/blob/main/google-cloud-bigquery/lib/google-cloud-bigquery.rb#L142), `config.add_field!` accepted `nil` as a valid value, although the validator was configured to `match: String`.

But with the change from [the PR](https://github.com/googleapis/google-cloud-ruby/pull/22866) `nil` is not considered valid anymore and, the following warning is outputted when the config is loaded:

```
Invalid value nil for key :endpoint. Setting anyway. at /.../gems/google-cloud-bigquery-1.44.0/lib/google-cloud-bigquery.rb:142:in `block in <main>'
```

Which basically means when `ENV["BIGQUERY_EMULATOR_HOST"]` is not set and the config should just use the default, it still would use the default but would always complain about that with a warning – which is a bit annoying.

As far as I can tell, allowing backwards compatibility with the previous behavior that didn't trigger this warning could be achieved by simply changing [line #142 in `google-cloud-ruby/blob/main/google-cloud-bigquery/lib/google-cloud-bigquery.rb`](https://github.com/googleapis/google-cloud-ruby/blob/main/google-cloud-bigquery/lib/google-cloud-bigquery.rb#L142) from

```ruby
config.add_field! :endpoint, default_endpoint, match: String
```

to

```ruby
config.add_field! :endpoint, default_endpoint, match: String, allow_nil: true
```

I am happy to create a PR for this fix, if that helps you. What do you think?

**Comment by alex-damian-negru** [comment]
Can confirm that this is a very annoying warning & a fix would be greatly appreciated. 🙂

**Comment by SandeepTuniki** [comment]
Thanks for spotting this, @spickermann.

> I am happy to create a PR for this fix, if that helps you. What do you think?

Yes, a PR would be greatly appreciated. I'll be happy to approve it.

**Comment by spickermann** [comment]
@SandeepTuniki Thanks for looking into this. I fix this annoying behavior in https://github.com/googleapis/google-cloud-ruby/pull/22904

**Comment by SandeepTuniki** [comment]
This fix is now released in version `1.44.1` of google-cloud-bigquery, through #22912

**Comment by spickermann** [comment]
> The fix is now released in version `1.44.1` of google-cloud-bigquery, through #22912

Thanks a lot @SandeepTuniki
